CHANGELOG.md

```diff
@@ -19,6 +19,8 @@ Our versioning strategy is as follows:
   * `import { editingDataService } from '@sitecore-jss/sitecore-jss-nextjs/editing';`
   * `import { EditingRenderMiddleware } from '@sitecore-jss/sitecore-jss-nextjs/editing';`
 
+* `[sitecore-jss-nextjs]` GraphqlSitemapService's constructor accepts a sites array by default, instead of siteName. fetchExportSitemap and fetchSSGSitemap methods will now return paths with site prefix applies by default. Provide a siteName parameter into the methods to get paths for a single site without prefix.
```

Comment:
Since we are now duplicating `graphql-sitemap-service.ts`, feel like we could we avoid the breaking change by an alternative design. 
1. Have both `siteName` (existing) and `sites` (new) as options on `GraphQLSitemapServiceConfig`, both of these are optional
2. Validate 1 (and only 1) of these are set in constructor
3. Base `next` template (single-site) will set `siteName` (as it does today); `nextjs-multisite` will set `sites`
4. Can then determine site prefix formatting based on config option used

Would avoid breaking change and might be more clear since determination is all based on service config vs combination of that + functions (w/ optional siteName param).

Reply:
> Have both siteName (existing) and sites (new) as options on GraphQLSitemapServiceConfig, both of these are optional
I have thought of several ways to combine multi and single site functionality, this one included. I think this would not be an optimal solution, even though it helps us avoid breaking changes. The entire behavior of a class changes based on constructor value. It makes using the class more complicated, less clean. If an instance of a class is passed several times between methods/other classes, you are never sure what behavior you will get without looking up the code.
I believe the approach of "Set default behavior for a class, modify behavior per method, via parameters" is cleaner, more intuitive. Multisite is a big change in itself, I think it could become the default way of using JSS going forward, and the breaking change here could be justified.

Reply:
That's the thing, it remains to be seen if multisite will be the norm. A rendering host per site is still valid and in many ways might be preferred. I see your point, but the same argument can be made for this current solution as well. It's confusing either way ☺. 

Wonder if we should really be introducing another service altogether for multisite (e.g. `GraphQLMultiSitemapService`)? Then it's 100% clear on what you're getting and APIs are catered towards each.

Reply:
Refactored it into two classes - please see if this is OK.

Reply:
I like it 👍. The two classes seems better to me. Not much duplication with the class extension.